### DISCUSSION TITLE
Part-DB 1.8.0 - REST API, Mouser info provider and more

### DISCUSSION THREAD

**Discussion by jbtronics** [Part-DB/Part-DB-server#410] [Announcements]
# Part-DB 1.8.0 - REST API, Mouser info provider and more

**If you like Part-DB, please consider donating to help further development. See the sponsor button on the [homepage](https://github.com/Part-DB/Part-DB-server).**

*Database schema changed: You need to run `php bin/console doctrine:migrations:migrate` after upgrade!* 

## New features
* Added an REST API, which allows external applications to access Part-DB. See [this announcement](https://github.com/Part-DB/Part-DB-server/discussions/393) and the [documentation](https://docs.part-db.de/api/intro.html) for more info.
* Added an information provider integration for mouser (thanks to @pdo59, see #329)
* Added a "new part" button to the navbar (#394)
* Allow to configure the default part table column visibility and order via an env option (thanks to @misaz, #390, #360)

## Bugfixes
* Fixed initial rendering of default currency entry in currency selectors
* Escape spaces in attachment URLs (#401)
* Show proper validation message if you are not allowed to create a new element using an entity selector instead of throwing an exception.

## Miscellaneous
* Updated dependencies
* Fixed typos in documentation
* 

## New Contributors
* @misaz made their first contribution in https://github.com/Part-DB/Part-DB-server/pull/390
* @tacman made their first contribution in https://github.com/Part-DB/Part-DB-server/pull/405
* @pdo59

**Full Changelog**: https://github.com/Part-DB/Part-DB-server/compare/v1.7.3...v1.8.0

<hr /><em>This discussion was created from the release <a href='https://github.com/Part-DB/Part-DB-server/releases/tag/v1.8.0'>Part-DB 1.8.0</a>.</em>

**Comment by w0hc** [comment]
Updated the docker, ran the migration with no errors. But trying to access the webpage I get:

Oops! An Error Occurred
The server returned a "500 Internal Server Error".
Something is broken. Please let us know what you were doing when this error occurred. We will fix it as soon as possible. Sorry for any inconvenience caused.

**Comment by tacman** [comment]
in .env, change APP_ENV from prod to dev, then you'll see a clearer error message.

**Comment by w0hc** [comment]
```
Symfony\Component\ErrorHandler\Error\ClassNotFoundError:
Attempted to load class "WebProfilerBundle" from namespace "Symfony\Bundle\WebProfilerBundle".
Did you forget a "use" statement for another namespace?

  at /var/www/html/vendor/symfony/framework-bundle/Kernel/MicroKernelTrait.php:126
  at App\Kernel->registerBundles()
     (/var/www/html/vendor/symfony/http-kernel/Kernel.php:346)
  at Symfony\Component\HttpKernel\Kernel->initializeBundles()
     (/var/www/html/vendor/symfony/http-kernel/Kernel.php:756)
  at Symfony\Component\HttpKernel\Kernel->preBoot()
     (/var/www/html/vendor/symfony/http-kernel/Kernel.php:185)
  at Symfony\Component\HttpKernel\Kernel->handle()
     (/var/www/html/vendor/symfony/runtime/Runner/Symfony/HttpKernelRunner.php:35)
  at Symfony\Component\Runtime\Runner\Symfony\HttpKernelRunner->run()
     (/var/www/html/vendor/autoload_runtime.php:29)
  at require_once('/var/www/html/vendor/autoload_runtime.php')
     (/var/www/html/public/index.php:5)       
```

**Comment by tacman** [comment]
Composer install

**Reply by w0hc** [reply]
That fixed it. Thank you.
